Update compatibility versions.

ingenthr opened the pull request:
Note I reviewed this with @brett19 and this is what we landed on.  Please review @AV25242 -- I know you were going to take this up after the meeting this morning, but there was just reason for Brett and I to do it on the fly so I could figure out what build dependencies we had.

ingenthr committed:
Update compatibility versions.

Note I reviewed this with @brett19 and this is what we landed on.  Please review @AV25242 -- I know you were going to take this up after the meeting this morning, but there was just reason for Brett and I to do it on the fly so I could figure out what build dependencies we had.

AV25242 reviewed (commented):
**should this be changed or be removed ?**
The Node.js SDK is built on top of libcouchbase (the C SDK), which is tested and supported on the following platforms:
**to** 
The Node.js SDK is built on top of couchbase++ , which is tested and supported on the following platforms:

**_Reason behind removed is that we dont publicly have anything called couchbase++ or c++ sdk_**

Also have the following questions 

1 - Are we saying we are dropping support for 

   CentOS 8, RHEL 7 ? 

2 - Are we not supporting Linux Alpine ? 

3 - On MacOS is it just 64bit or it also includes ARM if so can we explicitly say it ? 

4 -  For windows is this temporary because of the build issues ?

brett19 commented:
Hey @AV25242 ,
1 - Correct, these platforms are EOL
2 - We will support Alpine Linux
3 - We also will support M1
4 - Its worthwhile to differentiate "support", "works with" and "builds on" here.  You must use VS2022 to build on Windows, due to the compiler bug in MSVC 2019 we encountered, but that build should work fine on any windows system.

osfameron committed:
gardening

* reword to couchbase++
  (unless AV does decide to remove entirely - but if more platforms are moving
  to ++, this implementation detail will become visible in any case?)
* Note about ARM architecture for Mac.

Have not added Alpine note (unsure which version?)

osfameron commented:
@AV25242 pushed minor gardening commit - I've modified rather than removing the libcouchbase/couchbase++ statement, as I don't know if couchbase++ *will* be publicly exposed, or remain hidden? Happy to edit.

@brett19 which versions of Alpine Linux do we support?

AV25242 commented:
@osfameron  thanks much,  for all the changes
I am also not sure about couchbase++ i.e I know we are not allowing it for anyone to make any changes. Should we even call it out  ?  Why I am thinking it this way is because couchbase++ is not mentioned anywhere in our docs, to an end user it would be confusing unless we plan to have a short writeup about it somewhere ? 

Also when I read the following line 
The Node.js SDK 4.x is built on top of libcouchbase (the C SDK)couchbase++ library, _which is tested and supported on the following platforms:_
The immediate understanding is as if couchbase++ library is tested and supported not Node.js SDK , I think in this context. you mean to say Node.js SDK correct ? My recommendation would be to say 

The Node.js SDK 4.x is built on top of libcouchbase (the C SDK)couchbase++ library, and is supported on the following platforms:

osfameron commented:
@AV25242 yes, the libcouchbase -> Couchbase++ wording in docs has a few odd cases.
I've asked for clarification on some comments on https://issues.couchbase.com/browse/JSCBC-978 btw, if you have any thoughts!

I'll merge this for now (so it's in my working copy) and we can always come back to iterate on the wordings here!

osfameron committed:
Merge branch 'release/4.0' into versions-update

AV25242 commented:
so @osfameron  was in a conversation for Python and I brought this up IMHO couchbase++ is so not known that we should not have a mention of it (release notes is an exception) so I think we should go with this simple statement

**The Node.js SDK 4.x is supported on the following platforms:**. 

Thoughts ??

osfameron commented:
Yes, that's fine in that case - I've just folded that change back in in my branch (which I'll push shortly for review)

There are a few cases where we do link to the libcouchbase API docs for more details... 
I *think* for now I have to mention the libcouchbase API, but mention that the backend is actually Couchbase++, to avoid giving the wrong impression?

see e.g. https://docs.couchbase.com/nodejs-sdk/current/ref/client-settings.html

I'll fill in some suggested wording for now, and we can iterate!

AV25242 commented:
Yeah, that opens up a big question we dont have a Couchbase++ API, I think we will need to brainstorm. @raycardillo  is aware of it he is going to bring it up in the next meeting most likely.

AV25242 commented:
@osfameron going through the SDK Migration guide under the 4.x section I came across this

_**The required minimum version of Node.js is now v12**_

I think we should have this in our Compatibility page not just in migration page, this is an important consideration for someone to not just migrate but to also start using 4.x isn't it ?

RichardSmedley commented:
@AV25242 It's the opening paragraph of https://github.com/couchbase/docs-sdk-nodejs/blob/release/4.0/modules/project-docs/pages/compatibility.adoc  :+1:

AV25242 commented:
Thats perfect thanks @RichardSmedley  (I think I overlooked it my bad)

osfameron commented:
Yes indeed! I’ve also updated e.g. https://osfameron.github.io/DOC-9626-NodeJS-4-migration/nodejs-sdk/current/project-docs/compatibility.html with the latest numbers.

Did you spot other locations I’ve not updated? Do let me know and will fix!

--
Hakim Cassimally
Senior Technical Writer‌
***@***.***‌
‌
﻿﻿﻿﻿﻿[v63+PdS7y8wL78vdOPJZY8LLXUfwHZLxH6vUBIYgAAAABJRU5ErkJggg==]



From: Arun Vijayraghavan ***@***.***>
Reply to: couchbase/docs-sdk-nodejs ***@***.***>
Date: Wednesday, 16 February 2022 at 19:42
To: couchbase/docs-sdk-nodejs ***@***.***>
Cc: Hakim Cassimally ***@***.***>, Mention ***@***.***>
Subject: Re: [couchbase/docs-sdk-nodejs] Update compatibility versions. (PR #177)


@osfameron<https://github.com/osfameron> going through the SDK Migration guide under the 4.x section I came across this

The required minimum version of Node.js is now v12

I think we should have this in our Compatibility page not just in migration page, this is an important consideration for someone to not just migrate but to also start using 4.x isn't it ?

—
Reply to this email directly, view it on GitHub<https://github.com/couchbase/docs-sdk-nodejs/pull/177#issuecomment-1042096567>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAAG322ZURUDEVZTJCHQCQDU3P44DANCNFSM5OFQO6HQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
You are receiving this because you were mentioned.Message ID: ***@***.***>
